test/assert.js

```diff
@@ -2251,26 +2251,31 @@ describe('assert', function () {
         heroes = ['spiderman', 'superman'],
         fn     = function() { obj.value += 5 },
         fnDec  = function() { obj.value -= 20 },
+        getterFn = function() { return obj.value },
         bangFn = function() { obj.str += '!' },
         smFn   = function() { 'foo' + 'bar' },
         batFn  = function() { heroes.push('batman') },
         lenFn  = function() { return heroes.length };
 
     assert.changes(fn, obj, 'value');
+    assert.changes(fn, getterFn, 'value');
```

Comment:
I also think that this is incorrect.

When writing `assert.changes(fn, getterFn, 'value')` I understand that you expect `fn` to change the property `value` of `getterFn`.

It's redundant to specify a `key` if you already have a getter function. If we did this it would not be possible to check for property in `function` objects as well.

Reply:
I think the misunderstanding is located [here](https://github.com/chaijs/chai/issues/1090#issuecomment-345946379).
> I found it awkward when we use **assert(fn, getterFn, null, 1);**

Which should be `assert.changesBy(fn, getterFn, deltaValue);`

I will continue discussing on issue 1090, kindly re-validate with your point of view.

Reply:
Hi @brutalcrozt, thanks for the answer.

I'm sorry but I still didn't understand how this could be correct, since `delta` cannot be a `string`, which is what you're passing by using `'value'` as the last argument.

Can anyone else please review & confirm just for us to make sure we're not missing any detail?

Reply:
Hey brother, i think no need to ask third person point of view. I assume there is misunderstanding, i can say it correct just because run node test resulting green and travis also green. Have i show you [this](https://runkit.com/adi/5a16d0989cd06300128397b2) ?
Sorry i also make mistake again here, i mislead you. you are reviewing the `assert.changes` but i am comment about `assert.changesBy`.

---
I think that `assert.changes` pass because the key or property is be null when reaching this [line](https://github.com/chaijs/chai/blob/master/lib/chai/interface/assert.js#L2498).
do i need make additional change about this `assert.changes` ?
If you don't mind, please addressing `assert.changesBy` on other line. When you feel still necessary :smile: .

Reply:
Ah, I understand this now. You were right about the tests passing.

The reason why I was confused was because of [this line](https://github.com/chaijs/chai/blob/master/lib/chai/interface/assert.js#L2496), which uses the third argument as a message when the second one is a `function` and we only have three arguments.

However, would you mind just changing the `value` string to `example message` or something like that? Just to make sure the test looks clear.

Great work, btw! Thanks 😄

Reply:
Done, let me know if this still require some change.

---
Agreed, we should add it to next milestone. 4.2.0 ?
Kinda related to #1090, but unfortunately i still have issue to formulate the word. Let's continue there :wink: